ci/check_symbols.sh

```diff
@@ -47,17 +47,6 @@ for dso_file in ${dso_files}; do
     echo "  * WEAK: $(grep --count -E '  WEAK '    < ${symbol_file})"
     echo "  * LOCAL: $(grep --count -E '  LOCAL '  < ${symbol_file})"
 
-    echo "checking for 'fmt::' symbols..."
-    if grep -E 'fmt\:\:' < "${symbol_file}"; then
-        raise-symbols-found-error 'fmt::'
-    fi
-
-    echo "checking for 'spdlog::' symbols..."
-    if grep -E 'spdlog\:\:' < "${symbol_file}" \
-        | grep -v 'std\:\:_Destroy_aux'
-    then
-        raise-symbols-found-error 'spdlog::'
-    fi
```

Comment:
IMO we should keep this test here, at least during this period of transition while `rapids-logger` is changing so rapidly. 

That'd give us more confidence that `fmt` and `spdlog` symbols continue not to leak into `rmm`'s public interface, and I don't think this test is causing much maintenance burden (correct me if I'm wrong).

Reply:
https://github.com/rapidsai/rmm/pull/1808#discussion_r1943764489